### PR TITLE
restore: make incremental restore and log restore compatible

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/54474

Problem Summary:
1. current incremental backup is not compatible with log backup in one restore command. this pr try to make them compatible.

### What changed and how does it work?
1. skip cluster empty check when restore incremental + log backup.
2. provide an option for restore command to skip rewrite timestamp with a limitation ddl support.(NOTICE: this option will make incremental restore do a strict check for ddls by default)

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility
In order to protect data not loss for later log restore, the new add option during incremental restore will perform a strict check for ddls by default. which means the restore could fail due to unsupported backfill ddl types appeared. such as Add Index/Modify Column/Reorg Partition. 
Documentation https://github.com/pingcap/docs-cn/pull/17893/files

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that allow log restore perform based on an incremental restore.
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]
lgtm for ddl part

File: br/pkg/task/restore.go [hunk 1]
```diff
@@ -237,6 +239,10 @@ type RestoreConfig struct {
 	// if it is empty, directly take restoring log justly.
 	FullBackupStorage string `json:"full-backup-storage" toml:"full-backup-storage"`
 
+	// LogIncrementalCompat indicates whether this restore should enter a compatibility mode for incremental restore.
+	// In this restore mode, the restore will not perform rewrite on the incremental data.
+	LogIncrementalCompat bool `json:"log-incremental-compat" toml:"log-incremental-compat"`
```
Comment (BornChanger) [inline br/pkg/task/restore.go:244]:
```suggestion
	AllowPITRFromIncremental bool `json:"allow-pitr-from-incremental" toml:"allow-pitr-from-incremental"`
```

File: br/pkg/task/restore.go [hunk 2]
```diff
@@ -237,6 +239,10 @@ type RestoreConfig struct {
 	// if it is empty, directly take restoring log justly.
 	FullBackupStorage string `json:"full-backup-storage" toml:"full-backup-storage"`
 
+	// LogIncrementalCompat indicates whether this restore should enter a compatibility mode for incremental restore.
```
Comment (BornChanger) [inline br/pkg/task/restore.go:242]:
```suggestion
	// AllowPITRFromIncremental indicates whether this restore should enter a compatibility mode for incremental restore.
```

File: br/pkg/task/restore.go [hunk 3]
```diff
@@ -279,6 +285,10 @@ func DefineRestoreFlags(flags *pflag.FlagSet) {
 	_ = flags.MarkHidden(flagUseCheckpoint)
 
 	flags.Bool(FlagWaitTiFlashReady, false, "whether wait tiflash replica ready if tiflash exists")
+	flags.Bool(flagLogIncrementalCompat, true, "whether make incremental restore compatible with later log restore)"+
+		" if set to true, the incremental restore will not perform rewrite on the incremental data)"+
+		" meanwhile the incremental restore will not allow to restore 3 backfilled type ddl jobs,"+
+		" these restricted ddl jobs are add index, modify column and reorganize partition")
```
Comment (BornChanger) [inline br/pkg/task/restore.go:291]:
```suggestion
	flags.Bool(flagAllowPITRFromIncremental, true, "whether make incremental restore compatible with later pitr restore")
```

**Review by Leavrth — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**